src/python/pants/backend/python/goals/setup_py.py

```diff
@@ -563,72 +560,46 @@ async def generate_chroot(request: SetupPyChrootRequest) -> SetupPyChroot:
         }
     )
 
-    # Collect any `pex_binary` targets from `setup_py().with_binaries()`
-    key_to_binary_spec = exported_target.provides.binaries
-    binaries = await Get(
-        Targets, UnparsedAddressInputs(key_to_binary_spec.values(), owning_address=target.address)
-    )
-    entry_point_requests = []
-    for binary in binaries:
-        if not binary.has_field(PexEntryPointField):
-            raise InvalidEntryPoint(
-                "Expected addresses to `pex_binary` targets in `.with_binaries()` for the "
-                f"`provides` field for {exported_addr}, but found {binary.address} with target "
-                f"type {binary.alias}."
-            )
-        entry_point = binary[PexEntryPointField].value
-        url = "https://python-packaging.readthedocs.io/en/latest/command-line-scripts.html#the-console-scripts-entry-point"
-        if not entry_point.function:
-            raise InvalidEntryPoint(
-                "Every `pex_binary` used in `with_binaries()` for the `provides()` field for "
-                f"{exported_addr} must end in the format `:my_func` for the `entry_point` field, "
-                f"but {binary.address} set it to {entry_point.spec!r}. For example, set "
-                f"`entry_point='{entry_point.module}:main'. See {url}."
-            )
-        entry_point_requests.append(ResolvePexEntryPointRequest(binary[PexEntryPointField]))
-
-    binary_entry_points = await MultiGet(
-        Get(ResolvedPexEntryPoint, ResolvePexEntryPointRequest, request)
-        for request in entry_point_requests
-    )
-
-    entry_points_from_with_binaries = {
-        key: binary_entry_point.val.spec
-        for key, binary_entry_point in zip(key_to_binary_spec.keys(), binary_entry_points)
-        if binary_entry_point.val is not None
-    }
-
-    # Collect entry points from `python_distribution(entry_points=...)`
-    entry_points_from_field = {}
-    if exported_target.target.has_field(PythonDistributionEntryPointsField):
-        entry_points_field = await Get(
+    # Resolve entry points from python_distribution(entry_points=...) and from
+    # python_distribution(provides=setup_py(entry_points=...).with_binaries(...)
+    entry_points_field, provides_field = await MultiGet(
+        Get(
             ResolvedPythonDistributionEntryPoints,
             ResolvePythonDistributionEntryPointsRequest(
-                exported_target.target[PythonDistributionEntryPointsField]
+                entry_points_field=exported_target.target.get(PythonDistributionEntryPointsField)
             ),
-        )
```

Comment:
got to clean up a lot of code, as that functionality moved to resolve entry points rule.